src/seedu/addressbook/data/tag/Tagging.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.tag;
+
+import java.util.ArrayList;
+
+/*
+ * Association class;
+ * To print out all additions and removal of tags when the programme exits
+ */
+
+public class Tagging {
+
+    private ArrayList<String> allTagOpRecords = new ArrayList<>();
```

Comment:
Based on the class diagram given in the exercise, the list of `Tagging` objects should not be in this class. If it was in this class, the diagram would be like this:
![image](https://cloud.githubusercontent.com/assets/1673303/18409597/69111e4e-777d-11e6-92bf-9914c825bd13.png)